static/js/millennium_diff_table.js

```diff
@@ -0,0 +1,574 @@
+      if (resourceArray[i].type === resource.type) {
+        matchingResource = resourceArray[i];
+        delete resourceArray[i];
+      }
+    }
+  }
+  if (matchingResource != null) {
+    return {
+      dstu2Resources: [buildResource(resource)],
+      r4Resources: [buildResource(matchingResource)]
+    };
+  }
+  else {
+    return {
+      dstu2Resources: [buildResource(resource)]
+    };
+  }
+}
+
+/**
+ * Creates an object of matching DSTU2 and R4 resources that require special configuration.
+ * @param {object} config - The configuration object from resourceConfig.
+ * @param {array} dstu2Resources - The array of DSTU2 resource objects.
+ * @param {array} r4Resources - The array of R4 resource objects.
+ * @returns {object} Contains arrays of DSTU2 and R4 resources that match.
+ */
+function buildMatchFromConfig(config, dstu2Resources, r4Resources) {
+  let configuredResource = {
+    dstu2Resources: [],
+    r4Resources: [],
+  };
+
+  // Find the DSTU2 resources in the config
+  for (let i = 0; i < config.resourcesCombined.length; i++) {
+    for (let j = 0; j < dstu2Resources.length; j++) {
+      if (dstu2Resources[j] != null) {
+        if (dstu2Resources[j].type === config.resourcesCombined[i]) {
+          configuredResource.dstu2Resources.push(buildResource(dstu2Resources[j]));
+          delete dstu2Resources[j];
+        }
+      }
+    }
+  }
+
+  // Find the R4 resources in the config
+  for (let i = 0; i < config.combinedTo.length; i++) {
+    for (let j = 0; j < r4Resources.length; j++) {
+      if (r4Resources[j] != null) {
+        if (r4Resources[j].type === config.combinedTo[i]) {
+          configuredResource.r4Resources.push(buildResource(r4Resources[j]));
+          delete r4Resources[j];
+        }
+      }
+    }
+  }
+
+  if (config.hasOwnProperty('notes')) {
+    configuredResource.notes = config.notes;
+  }
+  return configuredResource;
+}
+
+/**
+ * Matches the resources between DSTU2 and R4.
+ * @param {object} conformanceData - The DSTU2 metadata.
+ * @param {object} capabilityStatementData - The R4 metadata.
+ * @returns {array} Array of all matching resource objects.
+ */
+function matchResources(conformanceData, capabilityStatementData) {
+  let dstu2Resources = conformanceData.rest[0].resource;
+  let r4Resources = capabilityStatementData.rest[0].resource;
+
+  let matchingResources = [];
+  let currentResource = null;
+  let config = null;
+  let match = null;
+
+  // Match all DSTU2 Resources
+  for (let i = 0; i < dstu2Resources.length; i++) {
+    currentResource = dstu2Resources[i];
+
+    if (currentResource != null) {
+      config = getConfiguration(currentResource.type);
+      match = null;
+
+      if (config != null) {
+        match = buildMatchFromConfig(config, dstu2Resources, r4Resources);
+        matchingResources.push(match);
+      }
+      else {
+        match = buildMatch(currentResource, r4Resources);
+        matchingResources.push(match);
+      }
+    }
+  }
```

Comment:
Curious if we can take this loop and the one on LN 206 and move them to their own functions to help seperate the logic out a bit further and make it easier to manage. The returned value from the functions would be an array and we could just combine the returned array with the one in the method.

Reply:
Updated here: https://github.com/cerner/fhir.cerner.com/pull/416/commits/de755cb18506fe0370836bfed9a96d57296d7080#diff-4066a5d03340f019ca6531e5bc567919R244-R256
https://github.com/cerner/fhir.cerner.com/pull/416/commits/de755cb18506fe0370836bfed9a96d57296d7080#diff-4066a5d03340f019ca6531e5bc567919R179-R201
https://github.com/cerner/fhir.cerner.com/pull/416/commits/de755cb18506fe0370836bfed9a96d57296d7080#diff-4066a5d03340f019ca6531e5bc567919R209-R234